mlrun/utils/helpers.py

```diff
@@ -1572,13 +1572,19 @@ def validate_component_version_compatibility(
     component_current_version = None
     try:
         if component_name == "iguazio":
-            parsed_current_version = mlrun.mlconf.get_parsed_igz_version()
             component_current_version = mlrun.mlconf.igz_version
+            parsed_current_version = mlrun.mlconf.get_parsed_igz_version()
+
+            # ignore pre-release versions and build metadata, as iguazio version is not semver compliant
```

Comment:
what is not semver compliant here?

Reply:
@liranbg `3.5.5-b687-202401182200` is smaller than `3.5.5` as `-` denotes pre-release. If we were to use `+` as `3.5.5+b687-202401182200` then the `b687-202401182200` would be semver compliant build metadata. According to https://semver.org/#spec-item-10